packages/flutter_test/lib/src/binding.dart

```diff
@@ -767,6 +773,12 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
     });
 
     return new Future<Null>.microtask(() async {
+      // Run all queued microtasks.
+      await new Future<Null>.microtask(() {});
+      // When the test had an exception, the test-framework already
+      // ran the teardown functions, removing the _fakeAsync function.
+      if (_fakeAsync == null)
+        return null;
```

Comment:
Is this really the cleanest way to solve this? I would hope that a refactor could find a much cleaner solution to this problem.